src/hb-aat-layout-feat-table.hh

```diff
@@ -172,6 +172,9 @@ struct feat
     return featureNameCount;
   }
 
+  bool exposes_feature (hb_aat_layout_feature_type_t feature_type) const
+  { return &get_feature (feature_type) != &Null(AAT::FeatureName); }
```

Comment:
Can this be turned to `{ return get_feature (feature_type).feature; }` we prefer not to compare with nullpool on possible cases which in here is possible AFAICS.

Reply:
We could do, but I'm a bit hesitant -- in principle, zero is a valid feature type constant. We don't currently map any tag to it (and it's unlikely that we ever will), but it still feels wrong to me. This method is supposed to be asking "does the 'feat' table have an entry for feature_type", and it should be valid to ask that for feature_type = 0.

Reply:
Correct. Anything else from the struct or combination of things which shouldn't be zero you believe? A has_data method can be added to FeatureName to write the logic instead hard coding it here.

Reply:
I guess we could assume that `nSettings` must be non-zero in any valid `FeatureName` record. (The spec doesn't explicitly say so, afaics, but having zero settings would be kinda meaningless, and would defeat the basic point of the 'feat' table to determine how to expose settings to the user.)